flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/Kafka010TableSink.java

```diff
@@ -51,11 +52,11 @@ public Kafka010TableSink(
 			String topic,
 			Properties properties,
 			SerializationSchema<Row> serializationSchema,
-			FlinkKafkaPartitioner<Row> partitioner) {
+			Optional<FlinkKafkaPartitioner<Row>> partitioner) {
 		return new FlinkKafkaProducer010<>(
 			topic,
 			serializationSchema,
 			properties,
-			partitioner);
+			partitioner.orElse(null));
```

Comment:
Why is the default here `null`, while for the JSON table sink it is the `FlinkFixedPartitioner`?

Reply:
The old Kafka producer had no optional. So `null` represents the default value for sinks.

Reply:
However, the deprecated JSON table sink used a `FlinkFixedPartitioner` as the default one and does not accept a null value.